src/PermitTokens/permitSignature.ts

```diff
@@ -309,9 +315,14 @@ export const addPermitTags = async (
       }
       const originalIndex = calls[i].tokenIndex;
       // add to existing token lists w tags for all tokens (permit or no permit)
-      if (!permitTokenInfo[originalIndex].tags)
-        (permitTokenInfo[originalIndex].tags as string[]) = [];
-      permitTokenInfo[originalIndex].tags!.push(tag);
+      const tags = permitTokenInfo[originalIndex].tags ?? [];
+
+      if (!permitTokenInfo[originalIndex].tags) {
+        (permitTokenInfo[originalIndex].tags as unknown as string[]) = [];
+      }
+
+      tags.push(tag);
+      (permitTokenInfo[originalIndex].tags as unknown as string[]) = tags;
```

Comment:
better do it without casting?

```suggestion
      if (!permitTokenInfo[originalIndex].tags) {
        (permitTokenInfo[originalIndex].tags as unknown as string[]) = [];
      }

      tags.push(tag);

      permitTokenInfo[originalIndex] = {
        ...permitTokenInfo[originalIndex],
        tags,
      };
```

Reply:
when `tags` is modified, `permitTokenInfo[originalIndex]` is also modified, so this would create duplicate

Reply:
Make sense